src/sdk/namespace_fs.js

```diff
@@ -654,7 +655,7 @@ class NamespaceFS {
         try {
             await this._load_bucket(params);
             for (const { key } of params.objects) {
-                const file_path = this._get_file_path({ key });
+                const file_path = this._get_file_path({ path: params.path, key });
```

Comment:
shouldn't the path come from the bucket config?

Reply:
according to our meeting, can I resolve this comment?

Reply:
I still don't feel comfortable with this approach of passing the path on every param when this is a property of the namespace itself...

Reply:
And I am still not sure why this.fs_path is not exactly the same as the params.path you used